src/lib.rs

```diff
@@ -630,6 +647,11 @@ impl<S: io::Read + io::Write> TlsStream<S> {
         Ok(self.0.peer_certificate()?.map(Certificate))
     }
 
+    /// Returns an iterator over certificate chain, if available.
+    pub fn certificate_chain(&mut self) -> Result<ChainIterator<S>> {
```

Comment:
Should we distinguish the "no chain present" case here by returning something like `Result<Option<ChainIterator>>`?

Reply:
I would prefer to create an empty iterator that just returns None immediately. That is what happens in the downstream crates. Result<Option<ChainIterator>> is a pretty annoying API.